lib/src/components/NxCombobox/__tests__/NxCombobox.test.tsx

```diff
@@ -0,0 +1,427 @@
+        inputElement = getByRole('combobox');
+
+    expect(inputElement.tagName).toBe('INPUT');
+    expect(inputElement).toHaveAttribute('role', 'combobox');
+  });
+
+  it('sets the value as the value of the input', function() {
+    const { getByRole } = rtlRender({ value: 'foo' }),
+        inputElement = getByRole('combobox');
+
+    expect(inputElement).toHaveValue('foo');
+  });
+
+  it('sets aria-autocomplete on the input to list as default and to both if `autoComplete` prop is set to true',
+      function() {
+        const { getByRole, rerender } = rtlRender(),
+            inputElement = getByRole('combobox');
+
+        expect(inputElement).toHaveAttribute('aria-autocomplete', 'list');
+
+        rerender(<NxCombobox {...minimalProps} autoComplete={true} />);
+
+        expect(inputElement).toHaveAttribute('aria-autocomplete', 'both');
+      });
+
+  it('sets a completion string of the selected suggestion from matches when `autoComplete` prop is set to true',
+      async function() {
+        const onChange = jest.fn(),
+            { getByRole } = rtlRender({ autoComplete: true,
+              matches: [{ id: '1', displayName: 'Foo' }, { id: '2', displayName: 'Boo' }],
+              onChange}),
+            inputElement = getByRole('combobox');
+
+        inputElement.focus();
+        await userEvent.type(inputElement, 'f');
+        expect(onChange).toBeCalledWith('Foo');
+      });
+
+  it('sets aria-expanded on the input to true when the dropdown displayed which has aria-hidden set to false',
+      async function() {
+        const { getByRole } = rtlRender({ matches: [{ id: '1', displayName: 'Foo' }] }),
+            inputElement = getByRole('combobox'),
+            dropdownElement = getByRole('listbox', { hidden: true });
+
+        expect(inputElement).toHaveAttribute('aria-expanded', 'false');
+        expect(dropdownElement).toHaveAttribute('aria-hidden', 'true');
+        await userEvent.type(inputElement, 'f');
+        expect(inputElement).toHaveAttribute('aria-expanded', 'true');
+        expect(dropdownElement).toHaveAttribute('aria-hidden', 'false');
+      });
+
+  it('calls onChange whenver the input\'s onChange event fires', async function() {
+    const onChange = jest.fn(),
+        { getByRole } = rtlRender({ onChange }),
+        inputElement = getByRole('combobox');
+
+    expect(onChange).not.toHaveBeenCalled();
+    await userEvent.type(inputElement, 'a');
+    expect(onChange).toHaveBeenCalledWith('a');
+  });
+
+  it('calls onSearch whenver the input\'s onChange event fires with a value that differs after trimming, ' +
+    'passing the trimmed value', async function() {
+    const onSearch = jest.fn(),
+        { getByRole } = rtlRender({ value: 'foo', onSearch }),
+        inputElement = getByRole('combobox');
+
+    expect(onSearch).not.toHaveBeenCalled();
+    await userEvent.type(inputElement, ' ');
+    expect(onSearch).not.toHaveBeenCalled();
+    await userEvent.type(inputElement, 'f');
```

Comment:
Isn't this testing that the user essentially typed "foo f"? If they do so we want to search for "foo f", not "foof"

Reply:
Thees two userEvents are not working together,
- line 116 is to test when user typed " ", and it won't call the `onSearch` since the value doesn't differ after trimming
- line 118 is to test when user typed "f", it will call the `onSearch` with "foof"

I updated the tests here to have the typed value " f", the `onSearch` should only be called one time with "foof". Hope this is more clear. 4d37f2e5b7d4c9ad0a973488db44c2925367926a.

Reply:
> I updated the tests here to have the typed value " f", the onSearch should only be called one time with "foof". Hope this is more clear. https://github.com/sonatype/sonatype-react-shared-components/commit/4d37f2e5b7d4c9ad0a973488db44c2925367926a.

Unfortunately it's not more clear.  In the first live example on the gallery page, if I enter "co", then go off somewhere else, then come back, click the box again and enter " l", it correctly tells like that no results were found.  This automated test case seems to be written as if it should be ignoring the space, searching for "col", and indicating that "colorado" is a match.

Reply:
As discussed and thanks for the suggestion, here is the updates 1aafa69fbe34426eaac9820c295f5d2c1d78c6ad.